Document how to use OIDC authentication (#967)
kubeapps/kubeapps: docs/user/login-alternatives.md

> # Kubeapps Login Form

Reviewer: perhaps this file should be `login-bypass.md`?

> By default, when first visiting Kubeapps, a login form is shown for the user to introduce a personal token for a specific service account:

Reviewer: we can just say a Kubernetes API token, it doesn't have to be a service account token

![Dashboard Login](../img/dashboard-login.png)

> You can find more information about access control in Kubeapps in this [document](./access-control.md).

Reviewer: This line doesn't feel very relevant here and it's already being linked to from other places, so I think we can skip it?

However, it's possible to disable this form or delegate the authentication to an OIDC provider so Kubeapps users don't need to introduce a token in the login form.

## Bypassing authentication

> Kubeapps expects an `Authorization` header that will be used to validate operations agains the Kubernetes API. This is usually set with the login form but if Kubeapps is exposed with an Ingress object, it's possible to hardcode a valid token in the Ingress configuration and automatically include it in all the requests.

Reviewer: ```suggestion
Kubeapps expects an `Authorization` header that will be used to validate operations against the Kubernetes API. This is usually set with the login form but if Kubeapps is exposed with an Ingress object, it's possible to hardcode a valid token in the Ingress configuration and automatically include it in all the requests.
```

> Kubeapps expects an `Authorization` header that will be used to validate operations agains the Kubernetes API. This is usually set with the login form but if Kubeapps is exposed with an Ingress object, it's possible to hardcode a valid token in the Ingress configuration and automatically include it in all the requests.

Reviewer: We could also rephrase this a bit, to make it more succinct:

```suggestion
Kubeapps will skip the login form if it detects that an `Authorization` header is already being set by a reverse-proxy, such as an Ingress. An Ingress object can be configured to inject a hardcoded valid Kubernetes API token on every request to Kubeapps to force the login form to be skipped and enable all API requests to use the hardcoded token.
```

> **NOTE**: This is not suitable for production since anyone with access to Kubeapps would be granted with the permissions associated with the hardcoded token.

Reviewer: ```suggestion
**NOTE**: This is not recommended in production since anyone with access to Kubeapps would be granted with the permissions associated with the hardcoded token.
```

I think "recommended" might be a better word to use here.

This is an example of the values that you can configure in the Kubeapps chart in order to set a valid token:

```yaml
ingress:
  enabled: true
  hosts:
    - name: kubeapps.local
      path: /
      tls: false
  annotations:
    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
    nginx.ingress.kubernetes.io/configuration-snippet: |
      add_header Authorization "Bearer TOKEN";
```

You just need to substitute TOKEN with the actual value of the token. The above assumes an Nginx [Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers), in case a different controller is being used that annotation will need to be adapted.

# Using an OIDC provider

> In case you want to use OAuth 2.0 to authenticate Kubeapps users follow this [guide](./using-an-OIDC-provider.md).

Reviewer: ```suggestion
In case you want to use OpenID Connect to authenticate Kubeapps users, and your Kubernetes API server is configured to use the same OIDC provider, follow this [guide](./using-an-OIDC-provider.md).
```
